jerry-core/ecma/operations/ecma-regexp-object.c

```diff
@@ -546,6 +565,7 @@ re_match_regexp (re_matcher_ctx_t *re_ctx_p, /**< RegExp matcher context */
 
         if (!ECMA_IS_VALUE_ERROR (match_value))
         {
+          REGEXP_RECURSION_COUNTER_INCREASE ();
```

Comment:
Why is this `REGEXP_RECURSION_COUNTER_INCREASE ();` different from the rest? All other macro calls are before a `return` but in this case a `return` can be reached without calling this macro.

Reply:
This was a misplaced macro. I replaced it right before the return statement.